### DISCUSSION TITLE
Additional options for drivers

### DISCUSSION THREAD

**Discussion by thiagomajesk** [FerretDB/FerretDB#4567] [General]
Hello! 👋 

I discovered this awesome project a few weeks ago while searching for viable MongoDB alternatives and I have a dumb question: Is there any plan to expose the DB in a different way other than through the wire protocol? 

> FerretDB is an open-source proxy that translates MongoDB wire protocol queries to SQL, with PostgreSQL or SQLite as the database engine. It uses the same commands, drivers, and tools as MongoDB.

This is very explicit in the project tagline, but I guess it doesn't hurt to ask... Having some compatibility with MongoDB is cool, but I wonder if someone has considered alternative ways of connecting to the database. For instance, something like CouchDB REST API comes to mind, but also, perhaps something like [MartenDB](https://martendb.io) that works directly with Postgres.

**Comment by AlekSi** [comment]
The implementation of the wire protocol is only a part of FerretDB; the implementation of MongoDB query semantics is another essential part. MongoDB has quite peculiar comparison rules, for example, that I don’t think would match the semantics of other databases. So, that seems unlikely as it is now. But we might reconsider that if there is enough demand.

**Reply by thiagomajesk** [reply]
Hi @AlekSi! Thanks for your reply... My question is more towards making it easier for library authors to create and maintain drivers for FerretDB (without changing query semantics and other features). 

For instance, MongoDB doesn't have a lot of adoption in the Elixir community, causing people who have a reason to use it to either reimplement the driver themselves or use unmaintained libraries. However, Postgres-based databases are much easier to get support for because we have an official Postgres driver. 

When I read about Marten and FerretDB I was like: this seems somewhat reasonable since FerretDB has a MongoDB compatibility layer, so getting rid of that could potentially make things even faster. 

Anyway, thanks for your time; again, this is truly an awesome project! 💪

**Reply by AlekSi** [reply]
Hmm, okay, that makes sense. Are there some existing HTTP APIs for MongoDB or MongoDB-like databases?

**Reply by thiagomajesk** [reply]
I think that Atlas supports something like this: https://www.mongodb.com/docs/atlas/app-services/data-api/ but I can't vouch for how good this middleware implementation actually is. The best reference I have for something like this is still CouchDB.

**Reply by AlekSi** [reply]
Thanks, we will investigate that

**Reply by AlekSi** [reply]
We are doing it (and some other things…): https://github.com/FerretDB/FerretDB/discussions/4578

**Reply by thiagomajesk** [reply]
Wow! So fast, that's great news! (perfect timing 😄)
